src/core/server/elasticsearch/client/client_config.ts

```diff
@@ -88,8 +88,9 @@ export function parseClientOptions(
         password: config.password,
       };
     } else if (config.serviceAccountToken) {
-      // TODO: change once ES client has native support for service account tokens: https://github.com/elastic/elasticsearch-js/issues/1477
-      clientOptions.headers!.authorization = `Bearer ${config.serviceAccountToken}`;
+      clientOptions.auth = {
```

Comment:
Not directly related to what this PR was supposed to address, but I saw this TODO and the client is now natively supporting bearer auth, so.